src/resolve/resolved-dns-transaction.c

```diff
@@ -2148,6 +2148,9 @@ int dns_transaction_request_dnssec_keys(DnsTransaction *t) {
                                         continue;
                         }
 
+                        if (t->scope->dnssec_mode == DNSSEC_NO)
+                                continue;
```

Comment:
this same check is already done at the beginning of the function so that the whole function is skipped when we are in the no-dnssec mode. Why check this here again?